src/labelformat/formats/maskpair.py

```diff
@@ -0,0 +1,211 @@
+
+
+@cli_register(format="maskpair", task=Task.INSTANCE_SEGMENTATION)
+class MaskPairInstanceSegmentationInput(InstanceSegmentationInput):
+    """Input format for image/mask pairs for instance segmentation.
+
+    This format loads images and corresponding binary masks, converts masks to
+    instance segmentations by finding connected components, and outputs in the
+    standard labelformat representation.
+    """
+
+    @staticmethod
+    def add_cli_arguments(parser: ArgumentParser) -> None:
+        parser.add_argument(
+            "--image-glob",
+            type=str,
+            required=True,
+            help="Glob pattern for image files (e.g., 'images/**/*.jpg')",
+        )
+        parser.add_argument(
+            "--mask-glob",
+            type=str,
+            required=True,
+            help="Glob pattern for mask files (e.g., 'masks/**/*.png')",
+        )
+        parser.add_argument(
+            "--base-path",
+            type=Path,
+            default=Path("."),
+            help="Base directory for glob patterns (default: current directory)",
+        )
+        parser.add_argument(
+            "--pairing-mode",
+            choices=["stem", "regex", "index"],
+            default="stem",
+            help="How to match images to masks: stem (filename), regex (numeric ID), index (sorted order)",
+        )
+        parser.add_argument(
+            "--category-names",
+            type=str,
+            required=True,
+            help="Comma-separated category names (e.g., 'crack,defect')",
+        )
+        parser.add_argument(
+            "--threshold",
+            type=int,
+            default=-1,
+            help="Binarization threshold 0-255, negative for automatic Otsu",
+        )
+        parser.add_argument(
+            "--min-area",
+            type=float,
+            default=10.0,
+            help="Minimum instance area in pixels to include",
+        )
+        parser.add_argument(
+            "--morph-open",
+            type=int,
+            default=0,
+            help="Morphological opening kernel size in pixels (0 to disable)",
+        )
+        parser.add_argument(
+            "--morph-close",
+            type=int,
+            default=0,
+            help="Morphological closing kernel size in pixels (0 to disable)",
+        )
+        parser.add_argument(
+            "--segmentation-type",
+            choices=["polygon", "rle"],
+            default="polygon",
+            help="Output segmentation format: polygon or RLE",
+        )
+        parser.add_argument(
+            "--approx-epsilon",
+            type=float,
+            default=0.0,
+            help="Polygon approximation factor (0.0 for full detail)",
+        )
+
+    def __init__(
+        self,
+        image_glob: str,
+        mask_glob: str,
+        base_path: Path = Path("."),
+        pairing_mode: str = "stem",
+        category_names: str = "object",
+        threshold: int = -1,
```

Comment:
Note that starting with threshold these are all just hyperparams. We could also simplify and drop them all and use a default value. They influence how we extract a polygon from the masked image. 
The PR allows to get both, polygons and RLE format. We can build up on top of the newly added code to convert RLE to polygon and vice versa.